src/transport/manager/mod.rs

```diff
@@ -1132,10 +1158,7 @@ impl TransportManager {
                                                         );
                                                         let _ = context
                                                             .tx
-                                                            .send(InnerTransportEvent::DialFailure {
-                                                                peer,
-                                                                addresses: vec![address.clone()],
-                                                            })
+                                                            .send(InnerTransportEvent::DialFailure { peer, addresses })
```

Comment:
@dharjeezy Can you also add to the doc of `InnerTransportEvent` & `TransportEvent` that `addresses` is only forwarded if the protocol was registered with `DialFailureAddresses::Required`, please?